### PR TITLE
WIP: Unify public key export

### PR DESCRIPTION
**Pull request by janmazak**
Cold keys and wallet keys in a single call.

### CODE REVIEW

File: src/bip44.h [hunk 1]
```diff
@@ -64,17 +62,41 @@ bool bip44_isValidStakingKeyPath(const bip44_path_t* pathSpec);
...
 
 size_t bip44_printToStr(const bip44_path_t*, char* out, size_t outSize);
 
 
+typedef enum {
+	// hd wallet account
+	PATH_WALLET_ACCOUNT,
+
+	// hd wallet address
+	PATH_WALLET_ADDRESS,
```
Comment (refi93) [inline src/bip44.h:79]: PATH_WALLET_SPENDING_KEY would be less ambiguous of a name, as technically, the staking key is also included in addresses

File: src/bip44.c [hunk 2]
```diff
@@ -260,6 +245,76 @@ size_t bip44_printToStr(const bip44_path_t* pathSpec, char* out, size_t outSize)
...
 }
 
+
+bip44_path_type_t bip44_classifyPath(const bip44_path_t* pathSpec)
+{
+	if (bip44_hasValidCardanoPoolColdKeyPrefix(pathSpec)) {
+		if (bip44_isValidPoolColdKeyPath(pathSpec)) {
+			return PATH_POOL_COLD_KEY;
+		} else {
+			return PATH_POOL_COLD_KEY_INVALID;
```
Comment (refi93) [inline src/bip44.c:255]: I don't see any code actually dealing with invalid paths on this granularity - I would just default to `PATH_INVALID` (renaming the PATH_UNKNOWN enum, so it's more accurate)

We can always introduce that kind of granularity if we would actually need it, but for now it seems like unnecessary bloat making this logic more complicated than it needs to be, unless I'm missing something

File: src/bip44.c [hunk 3]
```diff
@@ -260,6 +245,76 @@ size_t bip44_printToStr(const bip44_path_t* pathSpec, char* out, size_t outSize)
...
+
+		case 3: {
+			return PATH_WALLET_ACCOUNT;
+		}
+
+		case 5: {
+			const uint8_t chainType = bip44_getChainTypeValue(pathSpec);
+			switch (chainType) {
+
+			case 0:
```
Comment (refi93) [inline src/bip44.c:270]: don't we have proper enums for chain type?

File: src/securityPolicy.c [hunk 4]
```diff
@@ -624,41 +578,67 @@ security_policy_t policyForSignTxWitness(
...
+		switch (bip44_classifyPath(pathSpec)) {
 
-		} else {
+		case PATH_WALLET_ADDRESS:
+		case PATH_WALLET_STAKING_KEY:
+		case PATH_POOL_COLD_KEY:
+			if (bip44_isPathReasonable(pathSpec)) {
+				ALLOW();
+			} else {
+				WARN();
```
Comment (refi93) [inline src/securityPolicy.c:590]: this seems to be more strict than the original logic which allows any valid key - I find it a pointless to warn for unreasonable paths, especially if we don't return the public key - if the tx contains entries actually defined by such keys we already warn the user si warning them again during signing seems kind of redundant to me

Comment (refi93) [reply src/securityPolicy.c:590]: ah I see, the "WARN()" directive was embedded in the functions, so it's not really different, maybe apart from cold keys

Comment (janmazak) [reply src/securityPolicy.c:590]: The problem with witnesses is that they can be totally unrelated to the tx (no way to check). So whenever something suspicious occurs, it is better to warn the user.

**Comment by janmazak** [comment]
Merged into operatorapp.

**Pull request closed without merging**